pkg/k8s/cloudfactory.go

```diff
@@ -0,0 +1,197 @@
+package k8s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sts"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog"
+
+	"github.com/openshift/osd-utils-cli/cmd/common"
+	awsprovider "github.com/openshift/osd-utils-cli/pkg/provider/aws"
+)
+
+// CloudFactoryOptions defines the struct for running list account command
+type CloudFactoryOptions struct {
+	AccountName      string
+	AccountID        string
+	AccountNamespace string
+	ClusterID        string
+
+	Awscloudfactory awsprovider.CloudFactoryOptions
```

Comment:
I'm wondering if it would be less complicated to pull the aws stuff in separately inside each command, rather than do so inside this k8s factory?

Or if we did so here, the cloud factory would take a second argument of which cloud to provide defaults for?

so AttachCobraCliFlags would have `AttachCobraCLIFlags(cmd *cobra.Command, cloud string)` where `cloud` would be validated as one of `AWS`, or `GCP`, and then we populate that cloud providers' specific flags through that.

Otherwise, I think it would be more straightforward to do the following:

```go
k8sFactory.AttachCliOptions(cmd)
awsFactory.AttachCliOptions(cmd)
```

Reply:
We should make the k8s cloudfactory more flexible at some point. GCP account interaction will likely land here. We could go with this and then refactor later I suppose. At least something to think about

Reply:
The intention is that we shouldn't care which cloud we are using, we are asking for a cluster in most cases, which is getting its information from the k8s objects.

Ultimately I think we should add a layer above this which is `ocm`, so we ask `ocm` for cluster information and the tool works its way through the correct shards until it has what it needs. This may require some more careful design though.

Reply:
Could we rename it `CluserCloudFactory`?

Reply:
IMO the CloudFactory name doesn't make much sense here, bottom line what we want to manage here are accounts, shouldn't we call it `AccountFactory`? That way it could handle GCP accounts in the future too.

Reply:
After discussion with @rogbas we considered the name `ClusterResourceFactory`